internal/clients/sources/sources_impl.go

```diff
@@ -2,14 +2,68 @@ package sources
 
 import (
 	"context"
-
+	"encoding/json"
+	"fmt"
+	"github.com/RHEnVision/provisioning-backend/internal/cache"
 	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/parsing"
 )
 
+type AppType struct {
+	Id          string `json:"id"`
+	Name        string `json:"name"`
+	DisplayName string `json:"display_name"`
+}
+
+type dataElement struct {
+	Data []AppType `json:"data"`
+}
+
 func getSourcesClient(ctx context.Context) (SourcesIntegration, error) {
 	return NewClientWithResponses(config.Sources.URL)
 }
 
 func init() {
 	GetSourcesClient = getSourcesClient
 }
+
+func (c *ClientWithResponses) GetProvisioningTypeId(ctx context.Context, reqEditors ...RequestEditorFn) (string, error) {
+	if appTypeId, ok := cache.Cache.Get(cache.AppTypeKey); ok {
+		return appTypeId.(string), nil
+	}
+	if err := loadAppId(ctx, c); err != nil {
+		return "", err
+	}
```

Comment:
The function could have returned the key itself so additional lookups are not necessary, but it's ok.

Reply:
like this? :)